### PR TITLE
Consolidating some elevation angle code in FITACF 3.0

### PR DESCRIPTION
**Pull request by ecbland**
I found some code in FITACF 3.0 that's no longer needed now that FITACF 3.0 calls the separate `elevation` library. This PR removes that code. 

Key changes are:
- The `find_elevation()` function does both elevation calculations (i.e. from the lag zero phase and the fitted phase).
- Deleted the function `find_elevation_high()`, which previously did the elevation calculation from the fitted phase and was not calling the `elevation` library 
- Renamed the `find_elevation_low()` function to `find_elevation_error()`.
**Note: the least square error estimate is still based on the old elevation algorithm. I did not find a good way to integrate this calculation with the `elevation` library**
- Code comments updated

Please test this for a radar with a simple interferometer offset (Y-component only, e.g. SAS) and also for a radar with a more complicated offset (e.g. ZHO, LYR, HAN). 

```
make_fit -fitacf-version 3.0 [-old_elev] 20150308.1400.03.sas.rawacf > sas.fitacf3
dmapdump -d sas.fitacf3 | grep -A 5 '"elv"' | head -6
```

Comparing this branch with `develop`: 
- [ ] For radars with a simple Y-offset for the interferometer, output files should be identical on both branches
- [ ] For radars with a X or Z offset, there will be some small differences in the `elv_high` parameter (elevation calculated from the fitted phase)
- [ ] If the `-old_elev` option is used, the output files should be the same on both branches for all radars

### CODE REVIEW

**Comment by pasha-ponomarenko** [comment]
@ecbland, it seems to work as exected, i.e., no difference in elevation angle for radars with Y offset only, a very small difference (<0.3 deg) for radars with Y and Z offsets, and a larger difference (a degree or more) for radars with Y and X offsets. The largest difference is, of course, for ZHO (~10-20 deg).

Could you please eleborate a bit more on your third point:
_" If the -old_elev option is used, the output files should be the same for any interferometer layout."_

Do you mean that any changes to X or Z offsets will not change the output of the old algorithm as long as Y offset remains the same? 

There is a patch-up for Z offset in the old algorithm actually in determinations.c:

image
...
image

This is why the actual difference between the new and old algorithms for the radars with Z offset is very small.

**Comment by ecbland** [comment]
> @pasha-ponomarenko  Could you please eleborate a bit more on your third point:
> _" If the -old_elev option is used, the output files should be the same for any interferometer layout."_
> Do you mean that any changes to X or Z offsets will not change the output of the old algorithm as long as Y offset remains the same?

No, I will rephrase: If you use `make_fit -old_elev -fitacf-version 3.0`, the output file will be identical on this branch and the `develop` branch. This will be the case even for ZHO.

**Comment by pasha-ponomarenko** [comment]
@ecbland, thanks a lot for th eclarification! I was also thinking that it make sense to compare the outputs for the branches as well. I will try to test it ASAP.

**Comment by pasha-ponomarenko** [comment]
@ecbland, I processed some ZHO data (as a worst case scenario with large offsets in both X (-27.6) and Z (-5.3) directions) with this branch and _develop_ branch. While _elv_ and _elv-low_ fields were identical, there was **a significant difference in the _elv_high_ field for the _new_ algorithm** (i.e., processed without _-old_elev_ switch):

**_develop_ branch:**
ponomarenko@pasha-linux-main:~/work/elv_cleanup_test/devel_branch> dmapdump -d zho.fitacf3 | grep -A 5 '"elv"' | head -6
        float   "elv" [11]=
                22.008940,      4.723111,       38.451031,      6.324459,       7.418012,       8.472986,       19.313831,      4.995476,       24.209270,      36.372025,      36.491776
        float   "elv_low" [11]=
                0.520540,       0.335763,       0.358920,       0.208360,       0.533900,       0.254485,       0.170191,       0.208632,       0.955209,       0.621188,       0.566258
        **float   "elv_high" [11]=
                36.467842,      27.206762,      25.640871,      25.660721,      27.864281,      23.190001,      32.830238,      32.575184,      26.810225,      21.364899,      20.552984**
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

**this branch:**

ponomarenko@pasha-linux-main:~/work/elv_cleanup_test> dmapdump -d zho.fitacf3 | grep -A 5 '"elv"' | head -6
        float   "elv" [11]=
                22.008940,      4.723111,       38.451031,      6.324459,       7.418012,       8.472986,       19.313831,      4.995476,       24.209270,      36.372025,      36.491776
        float   "elv_low" [11]=
                0.520540,       0.335763,       0.358920,       0.208360,       0.533900,       0.254485,       0.170191,       0.208632,       0.955209,       0.621188,       0.566258
        **float   "elv_high" [11]=
                24.129257,      4.605854,       38.955143,      38.969410,      7.010735,       37.231865,      18.154741,      17.697992,      2.704549,       36.002575,      35.472023**
  
A noticeable difference in _elv_high_ values is also observed for INV (X offset is 1.5 m), and a much smaller but still detectable difference is present for HOK (Z offset is 2.9 m). 
Finally, results from SAS (no offsets in either X or Z directions) from these two branches are identical.

**In summary, the _new_ algorithm data are inconsistent between the two branches whenever there are offsets in X or/and Z directions.**

I will try to have a closer look at how the code has been modified.

**Comment by mts299** [comment]
I also got differences in `elv_high`

Here is what I ran
```
make_fit -fitacf-version 3.0 20180404.0200.12.lyr.rawacf > 20180404.0200.12.lyr.dev.elv.fitacf
```

old elevation worked fine. 

I compared it with `develop` and this branch

**Comment by pasha-ponomarenko** [comment]
@ecbland, @mts299, it looks like in the _develop_ branch the old elevation algorithm implemented (duplicated?) in _find_eleveation_high_ is used for calculating _elv_high_ regardless of the _elv_version_ value.   

In contrast, in this brach it is calculated with a corresponding version of the elevation algortihm:
image

In other words, in the _develop_ branch the elevation versioning has been implemented for calcualting _elv_normal_ in _find_elevation_ (i.e., using phi0) but the same thing has not been done for calculating _elv_high_ in _find_elevation_high_. This is illustrated by the fact that on the develop brach elv_high values without -old_elev swich are equal to those with this switch:

### develop branch results **with -old_elev switch**:

ponomarenko@pasha-linux-main:~/work/elv_cleanup_test/devel_branch> dmapdump -d **zho.fitacf3_old** | grep -A 5 '"elv"' | head -6
            float   "elv" [11]=
                    35.117908,      27.234852,      24.935209,      27.659449,      27.992153,      28.344744,      33.493519,      27.301697,      36.519905,      21.921192,      22.100010
            float   "elv_low" [11]=
                    0.520540,       0.335763,       0.358920,       0.208360,       0.533900,       0.254485,       0.170191,       0.208632,       0.955209,       0.621188,       0.566258
            **float   "elv_high" [11]=
                    36.467842,      27.206762,      25.640871,      25.660721,      27.864281,      23.190001,      32.830238,      32.575184,      26.810225,      21.364899,      20.552984**
     

### develop branch results **without -old_elev switch:**

ponomarenko@pasha-linux-main:~/work/elv_cleanup_test/devel_branch> dmapdump -d zho.fitacf3 | grep -A 5 '"elv"' | head -6
        float   "elv" [11]=
                22.008940,      4.723111,       38.451031,      6.324459,       7.418012,       8.472986,       19.313831,      4.995476,       24.209270,      36.372025,      36.491776
        float   "elv_low" [11]=
                0.520540,       0.335763,       0.358920,       0.208360,       0.533900,       0.254485,       0.170191,       0.208632,       0.955209,       0.621188,       0.566258
        **float   "elv_high" [11]=
                36.467842,      27.206762,      25.640871,      25.660721,      27.864281,      23.190001,      32.830238,      32.575184,      26.810225,      21.364899,      20.552984**


In summary, this branch seems to correct a previous omission as well! 
;-)

**Comment by ecbland** [comment]
> @pasha-ponomarenko In summary, this branch, besides refactoring, seems to correct a previous omission as well!

Yes you're correct. This is why the `elev_high` values are different on the two branches: `develop` was still using the old elevation calculation for `elev_high`. Sorry for not making this clearer in my original post. 

Notice in particular that `elv` and `elev_high` are very similar for ZHO on the new branch 😄

**Comment by pasha-ponomarenko** [comment]
> `develop` was still using the old elevation calculation for `elev_high`.

Thanks for the clarification -- it did not look like an intended behaviour! 
:-)
I guess this one is ready to go.

@mts299, what do you think?
